lib/flecks/slot.rb

```diff
@@ -3,21 +3,27 @@
 class Flecks::Slot < Phlex::HTML
 	include Phlex::DeferredRender
 
-	def initialize(promise)
+	def initialize(task)
+		task => Proc
 		@id = SecureRandom.urlsafe_base64
-		@promise = promise
+		@task = task
 		@placeholder = nil
+		@content = nil
 	end
 
 	def view_template
-		slot(name: @id, &@placeholder)
+		id = @id
+
+		slot(name: id, &@placeholder)
+		queue = context[:slots] ||= Flecks::Queue.new
+		queue.push(id, @task, @content)
```

Comment:
All we need to do here is queue these up and let the queue execute them asynchronously.